### PR TITLE
Allow hiding new/deleted formula/cask sections on update

### PR DESCRIPTION
**Pull request by eugenesvk**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/HEAD/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/HEAD/Library/Homebrew/test/PATH_spec.rb).
- [x] Have you successfully run `brew style` with your changes locally?
- [x] Have you successfully run `brew typecheck` with your changes locally?
- [ ] Have you successfully run `brew tests` with your changes locally? (see errors below)

-----

This information is mostly noise, so added an env var to allow to silence it

closes https://github.com/orgs/Homebrew/discussions/4590

### CODE REVIEW

File: docs/Manpage.md [hunk 1]
```diff
@@ -747,6 +747,8 @@ Fetch the newest version of Homebrew and all formulae from GitHub using `git`(1)
 * `-f`, `--force`:
   Always do a slower, full update check (even if unnecessary).
 
+If `HOMEBREW_NO_UPDATE_REPORT_FORMULAE`/`HOMEBREW_NO_UPDATE_REPORT_CASKS` is set, `brew update` will not show the list of newly added formulae/casks respectively
```
Comment (tyrann0us) [inline docs/Manpage.md:750]:
```suggestion
If `HOMEBREW_NO_UPDATE_REPORT_FORMULAE`/`HOMEBREW_NO_UPDATE_REPORT_CASKS` is set, `brew update` will not show the list of newly added formulae/casks, respectively.
```

**Review by tyrann0us — COMMENTED** [review]
Thanks for this! Does it also make sense to make disabling the _deleted_ formulae/casks lists possible?

**Comment by eugenesvk** [comment]
as far as I undersand, deleted section reports only those formulas that the user has installed, so it's more relevant information (unless `HOMEBREW_UPDATE_REPORT_ALL_FORMULAE` is set)

But maybe it still makes sense to silence those as well, so also added that option

File: Library/Homebrew/cmd/update-report.rb [hunk 2]
```diff
@@ -749,12 +749,12 @@ def dump(auto_update: false)
       report_all = false
     end
 
-    dump_new_formula_report
-    dump_new_cask_report
+    dump_new_formula_report if ENV["HOMEBREW_NO_UPDATE_REPORT_FORMULAE"].blank?
```
Comment (MikeMcQuaid) [inline Library/Homebrew/cmd/update-report.rb:752]: Define this in `env_config.rb` instead and use the generated `Homebrew.no_update_report*?` methods

File: docs/Manpage.md [hunk 3]
```diff
@@ -747,6 +747,9 @@ Fetch the newest version of Homebrew and all formulae from GitHub using `git`(1)
 * `-f`, `--force`:
   Always do a slower, full update check (even if unnecessary).
 
+If `HOMEBREW_NO_UPDATE_REPORT_FORMULAE`/`HOMEBREW_NO_UPDATE_REPORT_CASKS` is set, `brew update` will not show the list of newly added formulae/casks, respectively.
```
Comment (MikeMcQuaid) [inline docs/Manpage.md:750]: Define this in `env_config.rb` instead.

**Review by MikeMcQuaid — COMMENTED** [review]
@eugenesvk Thanks for the PR! 

Would a single variable that hid both meet both your needs?

@eugenesvk @tyrann0us

**Comment by MikeMcQuaid** [comment]
> Does it also make sense to make disabling the _deleted_ formulae/casks lists possible?

@tyrann0us Can you explain why you want to hide this? As @eugenesvk said, it's only those that you have installed so seems like pretty important information.

**Comment by eugenesvk** [comment]
A single var would be fine for me, just created 2 in case someone has more refined silencing tastes :)

**Comment by tyrann0us** [comment]
Hi @MikeMcQuaid,

> Would a single variable that hid both meet both your needs?

Yes, that would work fine for me. I'd actually prefer that.

>  Can you explain why you want to hide this? As @eugenesvk said, it's only those that you have installed so seems like pretty important information.

TBH, I wasn't aware that this section only lists deleted formulae/casks that the user has installed. 🙈
Totally agree that this is important information, so please disregard that comment. 

Thanks!

**Comment by MikeMcQuaid** [comment]
@eugenesvk Thanks. Do you personally want deleted formulae hidden or just new ones?

> TBH, I wasn't aware that this section only lists deleted formulae/casks that the user has installed. 🙈

Thanks. Could clarify the title of that section perhaps 👍🏻

File: docs/Manpage.md (hunk 3 above)
Comment (eugenesvk) [reply docs/Manpage.md:750]: `boolean:     true` there means variable type, not default value, correct?

File: Library/Homebrew/cmd/update-report.rb [hunk 4]
```diff
@@ -749,12 +749,12 @@ def dump(auto_update: false)
...
+    dump_new_cask_report if ENV["HOMEBREW_NO_UPDATE_REPORT_CASKS"].blank?
     dump_renamed_formula_report if report_all
     dump_renamed_cask_report if report_all
-    dump_deleted_formula_report(report_all)
-    dump_deleted_cask_report(report_all)
+    dump_deleted_formula_report(report_all) if ENV["HOMEBREW_NO_UPDATE_REPORT_FORMULAE_DELETED"].blank?
```
Comment (MikeMcQuaid) [inline Library/Homebrew/cmd/update-report.rb:756]: Given discussion: can remove this bit.

File: docs/Manpage.md (hunk 3 above)
Comment (eugenesvk) [reply docs/Manpage.md]: moved it to env_config

**Comment by eugenesvk** [comment]
> Do you personally want deleted formulae hidden or just new ones?
> 
> >

 Do you personally want deleted formulae hidden or just new ones?

No, to me it's important to know of some formula I have disappears. Personally I'm fine having a single var that only suppresses updates, and it seems you and @tyrann0us too, so will change to 1

File: Library/Homebrew/env_config.rb [hunk 5]
```diff
@@ -327,6 +327,22 @@ module EnvConfig
...
+        boolean:     true,
+      },
+      HOMEBREW_NO_UPDATE_REPORT_FORMULAE_DELETED:    {
+        description: "If set, `brew update` will not show the list of deleted formulae.",
+        boolean:     true,
+      },
+      HOMEBREW_NO_UPDATE_REPORT_CASKS_DELETED:    {
+        description: "If set, `brew update` will not show the list of deleted casks.",
+        boolean:     true,
+      },
```
Comment (MikeMcQuaid) [inline Library/Homebrew/env_config.rb:345]:
```suggestion
      HOMEBREW_NO_UPDATE_REPORT_NEW:    {
        description: "If set, `brew update` will not show the list of newly added formulae or casks.",
        boolean:     true,
      },
```

(may need to run `brew style --fix` and split the string if it's too long)

File: Library/Homebrew/cmd/update-report.rb [hunk 6]
```diff
@@ -749,12 +749,12 @@ def dump(auto_update: false)
       report_all = false
     end
 
-    dump_new_formula_report
-    dump_new_cask_report
+    dump_new_formula_report if ENV["HOMEBREW_NO_UPDATE_REPORT_FORMULAE"].blank?
+    dump_new_cask_report if ENV["HOMEBREW_NO_UPDATE_REPORT_CASKS"].blank?
```
Comment (MikeMcQuaid) [inline Library/Homebrew/cmd/update-report.rb:753]:
```suggestion
    if Homebrew::EnvConfig.no_update_report_new?
      dump_new_formula_report
      dump_new_cask_report
    end
```

File: Library/Homebrew/cmd/update-report.rb [hunk 7]
```diff
@@ -749,12 +749,12 @@ def dump(auto_update: false)
...
     dump_renamed_formula_report if report_all
     dump_renamed_cask_report if report_all
-    dump_deleted_formula_report(report_all)
-    dump_deleted_cask_report(report_all)
+    dump_deleted_formula_report(report_all) if ENV["HOMEBREW_NO_UPDATE_REPORT_FORMULAE_DELETED"].blank?
+    dump_deleted_cask_report(report_all) if ENV["HOMEBREW_NO_UPDATE_REPORT_CASKS_DELETED"].blank?
```
Comment (MikeMcQuaid) [inline Library/Homebrew/cmd/update-report.rb:757]:
```suggestion
    dump_deleted_formula_report(report_all)
    dump_deleted_cask_report(report_all)
```

File: Library/Homebrew/cmd/update-report.rb (hunk 6 above)
Comment (MikeMcQuaid) [reply Library/Homebrew/cmd/update-report.rb]: @eugenesvk check out this suggestion

File: Library/Homebrew/cmd/update-report.rb (hunk 4 above)
Comment (eugenesvk) [reply Library/Homebrew/cmd/update-report.rb]: removed

File: Library/Homebrew/cmd/update-report.rb (hunk 7 above)
Comment (eugenesvk) [reply Library/Homebrew/cmd/update-report.rb]: sry missed this change, already added

File: Library/Homebrew/cmd/update-report.rb (hunk 6 above)
Comment (eugenesvk) [reply Library/Homebrew/cmd/update-report.rb]: yes, the typed env is much nicer, thanks

**Review by MikeMcQuaid — APPROVED** [review]
Thanks so much for your repeated contribution! Without people like you submitting PRs we couldn't run this project. You rock, @eugenesvk!

**Pull request merged**